contracts/PurchaseHook.sol

```diff
@@ -4,31 +4,32 @@ pragma solidity ^0.8.0;
 import "@unlock-protocol/contracts/dist/PublicLock/IPublicLockV13.sol";
 import "hardhat/console.sol";
 
+error Unauthorized();
+
 contract PurchaseHook {
+    IPublicLockV13 lock;
     mapping(address => address payable) public referals;
     mapping(address => uint256) public referalAmounts;
 
     /** Constructor */
-    constructor() {}
+    constructor(IPublicLockV13 _lock) {
```

Comment:
I added the lock address to the constructor to make sure only a lock manager of that lock can set referrer amounts.